llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -4474,7 +4474,7 @@ void SelectionDAGBuilder::visitAlloca(const AllocaInst &I) {
 
   // Mask out the low bits for alignment purposes.
   AllocSize = DAG.getNode(ISD::AND, dl, AllocSize.getValueType(), AllocSize,
-                          DAG.getConstant(~StackAlignMask, dl, IntPtr));
+                          DAG.getSignedConstant(~StackAlignMask, dl, IntPtr));
```

Comment:
Can we use DAG.getZeroExtendInReg here?

Reply:
I think we'd have to convert StackAlign to an EVT?

Reply:
Don't worry about it - my mind was wondering - we want to mask out the lower bits, not zero the upper bits. Sorry for the noise.